controllers/component_build_controller_pac.go

```diff
@@ -483,63 +479,38 @@ func (r *ComponentBuildReconciler) UndoPaCProvisionForComponent(ctx context.Cont
 
 func (r *ComponentBuildReconciler) lookupPaCSecret(ctx context.Context, component *appstudiov1alpha1.Component, gitProvider string) (*corev1.Secret, error) {
 	log := ctrllog.FromContext(ctx)
-	sourceUrl := component.Spec.Source.GitSource.URL
-
-	url, err := url.Parse(sourceUrl)
-	if err != nil {
-		r.EventRecorder.Event(&corev1.Secret{}, "Warning", "ErrorParsingPACRepo", err.Error())
-		return nil, fmt.Errorf("failed to parse component's source URL: %w", err)
+	gitSource := component.Spec.Source.GitSource
+	url := strings.TrimSuffix(strings.TrimSuffix(gitSource.URL, ".git"), "/")
+	branch := gitSource.Revision
+	if branch == "" {
+		branch = InternalDefaultBranch
 	}
-	componentRepo, _ := strings.CutSuffix(url.Path, ".git")
-
-	log.Info("Looking for Pipelines as Code SCM secret", "host", url.Hostname(), "repo", componentRepo)
-
-	secretList := &corev1.SecretList{}
-	opts := client.ListOption(&client.MatchingLabels{
-		scmCredentialsSecretLabel: "scm",
-		scmSecretHostnameLabel:    url.Hostname(),
-	})
-
-	if err := r.Client.List(ctx, secretList, client.InNamespace(component.Namespace), opts); err != nil {
-		r.EventRecorder.Event(&corev1.Secret{}, "Warning", "ErrorListingPaCSecrets", err.Error())
-		return nil, fmt.Errorf("failed to list Pipelines as Code secrets in %s namespace: %w", component.Namespace, err)
+	scmComponent := git.ScmComponent{
+		NamespaceName: component.Namespace,
+		ComponentName: component.Name,
+		RepositoryUrl: url,
```

Comment:
In some places normalized URL and host are used. Does it make sense to have a constructor and corresponding fields, WDYT?